### ISSUE TITLE
neofs stops working and eats all CPU

### ISSUE THREAD

**Issue by 532910** [nspcc-dev/neofs-node#1042]
neofs stops working on niko some time after start:
```
2021-12-13T22:25:23.879Z        info        neofs-node/grpc.go:82        start listening gRPC endpoint        {"endpoint": "10.78.20.11:8080"}
2021-12-13T22:25:23.879Z        info        neofs-node/grpc.go:82        start listening gRPC endpoint        {"endpoint": "10.78.20.11:8082"}
2021-12-13T22:25:24.131Z        info        replicator/process.go:20        process routine        {"component": "Object Replicator", "task queue capacity": 0, "put timeout": 5}
2021-12-13T22:25:24.256Z        info        neofs-node/main.go:99        application started        {"build_time": "2021-12-09T12:04:11+00:00", "version": "v0.27.0", "debug": "false"}
2021-12-14T10:10:29.954Z        warn        subscriber/subscriber.go:134        remote notification channel has been closed
2021-12-14T10:10:29.955Z        warn        event/listener.go:236        stop event listener by notification channel
2021-12-14T10:10:29.955Z        warn        replicator/process.go:28        context is done        {"component": "Object Replicator", "error": "context canceled"}
2021-12-14T10:10:29.955Z        info        replicator/process.go:15        routine stopped        {"component": "Object Replicator"}
2021-12-14T10:10:29.955Z        warn        neofs-node/main.go:111        internal application error        {"message": "event subscriber connection has been terminated"}
2021-12-14T10:10:29.955Z        info        neofs-node/config.go:358        closing components of the storage engine...
2021-12-14T10:10:29.961Z        info        neofs-node/config.go:366        all components of the storage engine closed successfully
2021-12-14T10:10:29.961Z        info        neofs-node/grpc.go:96        stopping gRPC server...        {"name": "NeoFS Public API"}
2021-12-14T10:10:30.072Z        info        neofs-node/grpc.go:100        gRPC server stopped successfully        {"name": "NeoFS Public API"}
2021-12-14T10:10:30.072Z        info        neofs-node/grpc.go:96        stopping gRPC server...        {"name": "NeoFS Public API"}
2021-12-14T10:10:30.072Z        info        neofs-node/grpc.go:100        gRPC server stopped successfully        {"name": "NeoFS Public API"}
2021-12-14T10:10:30.072Z        info        neofs-node/grpc.go:75        stop listening gRPC endpoint        {"endpoint": "10.78.20.11:8080"}
2021-12-14T10:10:30.072Z        info        neofs-node/grpc.go:75        stop listening gRPC endpoint        {"endpoint": "10.78.20.11:8082"}
2021-12-14T10:10:30.092Z        info        neofs-node/grpc.go:96        stopping gRPC server...        {"name": "NeoFS Control API"}
2021-12-14T10:10:30.092Z        info        neofs-node/grpc.go:100        gRPC server stopped successfully        {"name": "NeoFS Control API"}
```
It also closes the monitoring port and eats 100% of the CPU.

```
$ neofs-node --version
Version: v0.27.0 
Build: 2021-12-09T12:04:11+00:00 
Debug: false
```

I restarted it with debug logging and will update this issues as soon as I found it broken.

**Comment by alexvanin** [comment]
The issue is that neo-go RPC node closes web socket connection with neofs node:
```
warn        subscriber/subscriber.go:134        remote notification channel has been closed
warn        event/listener.go:236        stop event listener by notification channel
warn        neofs-node/main.go:111        internal application error        {"message": "event subscriber connection has been terminated"}
```

It triggers application termination. Check if RPC node is healthy or try another one. 

During application initialization and termination it is expected to see some CPU load spikes.

**Comment by 532910** [comment]
![image](https://user-images.githubusercontent.com/14617699/146137106-cad581e3-2e22-44a6-b5f2-3375a095cf89.png)

**Comment by 532910** [comment]
niko uses mainnet rpc nodes

**Comment by alexvanin** [comment]
We did fixes for v0.27.2+
- https://github.com/nspcc-dev/neofs-node/pull/1047
- https://github.com/nspcc-dev/neofs-api-go/pull/366

It should be okay now. If neofs-node still freezes over time and eats all CPU, reopen the issue.
